docs: add CDK examples (#113)
mrgrain/jsii-struct-builder: README.md

npm install --save-dev @mrgrain/jsii-struct-builder
```

> Then, place a `new ProjenStruct` in your `.projenrc.js` file, passing your
> [TypeScript project](https://projen.io/typescript.html) as the first parameter. See the sections below for more usage
> details.

Reviewer: ```suggestion
Then add a `new ProjenStruct` in your `.projenrc.ts` file, passing a [TypeScript project](https://projen.io/typescript.html) as the first parameter.
See the sections below for more usage details.
```

Reviewer: Yes!

https://projen.io/typescript.html#migrating-projenrcjs-to-projenrcts

New projects are setup with it by default! Looks like your projen version is out of date 😬 
```console
$ npx projen@latest new awscdk-construct
Need to install the following packages:
  projen@0.71.91
Ok to proceed? (y) y
👾 Project definition file was created at /Users/kornherm/code/scratchpad/pts/.projenrc.ts
```

> If you're not using [`projen`](https://projen.io/), see [Use without `projen`](#use-without-projen), below.

Reviewer: ```suggestion
Or if you're not using [`projen`](https://projen.io/), see [Use without `projen`](#use-without-projen).
```

Any existing properties of the same name will be replaced.

```ts

> new ProjenStruct(project, { name: 'MyProjectOptions' })

Author: `prettier` made some changes - lmk if you want me to back them out

  .withoutDeprecated();
```

### AWS CDK properties

A common use-case of this project is to expose arbitrary overrides in CDK constructs.
For example, you may want to provide common AWS Lambda configuration, but allow a consuming user to override any arbitrary property.

To accomplish this, first create the new struct in your `.projenrc.ts` file.

> ```js

Reviewer: ```suggestion
```ts
```

  .withoutDeprecated();
```

> ### Exposing CDK Properties
> 
> A common use-case for this project is to expose arbitrary overrides in CDK constructs. For example, you may want to
> provide common AWS Lambda configuration, but allow a consuming user to override any arbitrary property.
> 
> To accomplish this, first create the new struct in your `.projenrc.js` file.

Reviewer: ```suggestion
### AWS CDK properties

A common use-case of this project is to expose arbitrary overrides in CDK constructs.
For example, you may want to provide common AWS Lambda configuration, but allow a consuming user to override any arbitrary property.

To accomplish this, first create the new struct in your `.projenrc.ts` file.
```

```js
const { ProjenStruct, Struct } = require('@mrgrain/jsii-struct-builder');
const { awscdk } = require('projen');

const project = new awscdk.AwsCdkConstructLibrary({
  // your config - see https://projen.io/awscdk-construct.html
});

new ProjenStruct(project, { name: 'MyFunctionProps' })
  .mixin(Struct.fromFqn('aws-cdk-lib.aws_lambda.FunctionProps'))
  .withoutDeprecated()
  .allOptional()

>   .omit('code'); // our construct always provides the code

Reviewer: ```
```ts
import { ProjenStruct, Struct } from '@mrgrain/jsii-struct-builder';
import { awscdk } from 'projen';

declare project: awscdk.AwsCdkConstructLibrary;

new ProjenStruct(project, { name: 'MyFunctionProps' })
  .mixin(Struct.fromFqn('aws-cdk-lib.aws_lambda.FunctionProps'))
  .withoutDeprecated()
  .allOptional()
  .omit('code'); // our construct always provides the code
```

Author: See comment above, RE: ts projen config file

> ### Exposing CDK Properties
> 
> A common use-case for this project is to expose arbitrary overrides in CDK constructs. For example, you may want to
> provide common AWS Lambda configuration, but allow a consuming user to override any arbitrary property.
> 
> To accomplish this, first create the new struct in your `.projenrc.js` file.
> 
> ```js
> const { ProjenStruct, Struct } = require('@mrgrain/jsii-struct-builder');
> const { awscdk } = require('projen');
> 
> const project = new awscdk.AwsCdkConstructLibrary({
>   // your config - see https://projen.io/awscdk-construct.html
> });
> 
> new ProjenStruct(project, { name: 'MyFunctionProps' })
>   .mixin(Struct.fromFqn('aws-cdk-lib.aws_lambda.FunctionProps'))
>   .withoutDeprecated()
>   .allOptional()
>   .omit('code'); // our construct always provides the code
> ```
> 
> Then, expose the new struct in your CDK construct.

Reviewer: ```suggestion
### AWS CDK properties

A common use-case of this project is to expose arbitrary overrides in CDK constructs.
For example, you may want to provide common AWS Lambda configuration, but allow a consuming user to override any arbitrary property.

To accomplish this, first create the new struct in your `.projenrc.ts` file.

```ts
import { ProjenStruct, Struct } from '@mrgrain/jsii-struct-builder';
import { awscdk } from 'projen';

declare project: awscdk.AwsCdkConstructLibrary;

new ProjenStruct(project, { name: 'MyFunctionProps' })
  .mixin(Struct.fromFqn('aws-cdk-lib.aws_lambda.FunctionProps'))
  .withoutDeprecated()
  .allOptional()
  .omit('code'); // our construct always provides the code
```

Then, expose the new struct in your CDK construct:
```

> Then, expose the new struct in your CDK construct.

Reviewer: ```suggestion
Then use the new struct in your CDK construct.
```
